scipy/ndimage/src/_ni_measure.pyx

```diff
@@ -0,0 +1,537 @@
+######################################################################
+# Cython version of scipy.ndimage.measurement.find_objects() function.
+######################################################################
+
+cimport cython
+from cython cimport sizeof
+import numpy as np
+cimport numpy as np
+
+np.import_array()
+
+cdef extern from *:
+   ctypedef int Py_intptr_t
+
+cdef extern from "numpy/arrayobject.h" nogil:
+    ctypedef struct PyArrayIterObject:
+        np.intp_t *coordinates
+        char *dataptr
+        np.npy_bool contiguous
+
+    void *PyDataMem_NEW(size_t)
+    void PyDataMem_FREE(void *)
+
+######################################################################
+# Use of Cython's type templates for type specialization
+######################################################################
+
+# Only integer values are allowed.
+ctypedef fused data_t:
+    np.int8_t
+    np.int16_t
+    np.int32_t
+    np.int64_t
+    np.uint8_t
+    np.uint16_t
+    np.uint32_t
+    np.uint64_t
+
+######################################################################
+# Update Regions According to Input Data Type
+######################################################################
+
+cdef inline np.ndarray arr_corr(np.ndarray raw_array, int flags):
+    cdef np.ndarray array
+    if raw_array is not None:
+        array = np.PyArray_FROM_OF(raw_array, flags)
+    else:
+        array = raw_array
+    return array
+
+cdef inline findObjectsPoint(data_t *data, np.flatiter _iti, np.intp_t max_label, 
```

Comment:
Instead of `np.flatiter _iti`, you could accept a `np.intp_t *cc`, and then call this function with `iti.coordinates`.